### ISSUE TITLE
WurstSetup takes too long boot time. Here are some research & feed back.

### ISSUE THREAD

**Issue by LazyKnightX** [wurstscript/WurstSetup#13]
Hi.

I'm using Wurst as my Warcraft 3 script language, I can connect Google directly quickly.
But when I execute WurstSteup.jar, it always takes more than 1 minute to boot up and without any information. (I've finished setup wurst and working with it, but it takes too long time to boot.)
I checked the connection, WurstSteup.jar connect to Google first, maybe you're using services like Google Analytics, but why that connect cost time like this?

And is there any way I can try to decrease the boot time?

Thank you for any hint, thanks!

**Comment by LazyKnightX** [comment]
Update.

After my friend's research, we find it connected to google first for test network connectivity.
It's fine, but it seems first ping to "google.com", then "www.google.com". And after the time-out, it ping to "www.google.com" again.
I shut down the connection to "www.google.com", the setup booted, and service is online.

Seem there are some reasons you re-connect google for multiple times.
But could we manual close this feature? or could you make users can change URL (like www.microsoft.com) to check the ping?

Thank you for reply.

**Comment by LazyKnightX** [comment]
Connect status:

![](https://i.imgur.com/SFYF6OS.png)

**Comment by PhoenixZeng** [comment]
sample change the
`net/ConnectionManager.kt : 37` to `http://www.microsoft.com/`
can solve this problem. 
before: 60s to run
after:9s to run

**Comment by LazyKnightX** [comment]
Personally, I hope they could provide the offline installer, this will help us to deliver it in none English environment and make more people know this great language.

**Comment by Frotty** [comment]
It's an online and compatibility check.
The only reason google is so bad for you is because your government sucks I guess? ;)
But yeah, I suppose it could have 2-3 endpoints and as soon as one responds it's enough.
Also the UI proably shouldn't wait for it.

**Comment by LazyKnightX** [comment]
@Frotty 
Looking forward to the update, thank you!
XD

**Comment by peq** [comment]
> Personally, I hope they could provide the offline installer, this will help us to deliver it in none English environment and make more people know this great language.

If you don't want to use the installer, you can find direct downloads (e.g. `wurstpack_complete.zip`)  at https://grill.wurstlang.org/hudson/job/Wurst/lastSuccessfulBuild/artifact/downloads/

**Comment by LazyKnightX** [comment]
> If you don't want to use the installer, you can find direct downloads (e.g. wurstpack_complete.zip) at https://grill.wurstlang.org/hudson/job/Wurst/lastSuccessfulBuild/artifact/downloads/

Thank you for providing this, it seems works fine.

**Comment by Frotty** [comment]
Nah, it doesn't "install" wurst. You still have to do all other steps manually and also link the compiler yourself.
I think a offline project/compiler install is a good idea.

**Comment by LazyKnightX** [comment]
@Frotty That will be good if you provide the offline project/compiler installer.
I'm continuing keep my attention on this.
Thank you for reopen this.

**Comment by Frotty** [comment]
I will make a new ticket for the offline functionality.
The online check is now running async, allowing the UI to load earlier.
I might still do the "fastest of two sources" at some point.
